typescript/radix-sdk/src/ism/ism-query.ts

```diff
@@ -0,0 +1,186 @@
+import { GatewayApiClient } from '@radixdlt/babylon-gateway-api-sdk';
+
+import { assert, ensure0x } from '@hyperlane-xyz/utils';
+
+import {
+  getComponentOwner,
+  getComponentState,
+  getFieldElementsFromEntityState,
+  getFieldValueFromEntityState,
+  getKeysFromKvStore,
+  getRadixComponentDetails,
+} from '../utils/base-query.js';
+import {
+  EntityField,
+  MultisigIsms,
+  RadixElement,
+  RadixIsmTypes,
+} from '../utils/types.js';
+
+function isIsmType(maybeIsmType: string): maybeIsmType is RadixIsmTypes {
+  switch (maybeIsmType) {
+    case RadixIsmTypes.MERKLE_ROOT_MULTISIG:
+    case RadixIsmTypes.MESSAGE_ID_MULTISIG:
+    case RadixIsmTypes.NOOP_ISM:
+    case RadixIsmTypes.ROUTING_ISM:
+      return true;
+  }
+
+  return false;
+}
+
+export async function getIsmType(
+  gateway: Readonly<GatewayApiClient>,
+  ismAddress: string,
+): Promise<RadixIsmTypes> {
+  const ismDetails = await getRadixComponentDetails(gateway, ismAddress, 'ISM');
+
+  const ismType = ismDetails.blueprint_name;
+  assert(
+    isIsmType(ismType),
+    `Expected component at address ${ismAddress} to be an ism but got ${ismType}`,
+  );
+
+  return ismType;
+}
+
+export async function getTestIsmConfig(
+  gateway: Readonly<GatewayApiClient>,
+  ismAddress: string,
+): Promise<{
+  type: RadixIsmTypes.NOOP_ISM;
+  address: string;
+}> {
+  const ismDetails = await getRadixComponentDetails(gateway, ismAddress, 'ISM');
```

Comment:
nit: i realized 'ISM' is used in the logging, should it be more specific i.e. RadixIsmTypes.NOOP_ISM?